src/vs/editor/contrib/hover/browser/markdownHoverParticipant.ts

```diff
@@ -44,16 +57,26 @@ export class MarkdownHover implements IHoverPart {
 	}
 }
 
+interface HoverSourceInformation {
+	readonly hover: Hover;
```

Comment:
The hover is disposable. Who owns (i.e. disposes) it? The code really looks like there could be a leak...

Reply:
So the HoverSource is sent as part of the MarkdownHover inside of the computeAsync method. Then it is sent back in the hoverParts parameter inside of the renderHoverParts method. 

The renderHoverParts method will ultimately add the hover into the `renderedHoverPart.disposables` object.

If the HoverSource object is undefined then Hover should presumably not exist.

I am thinking of a way to make the disposal clearer